LDM-633.tex

```diff
@@ -0,0 +1,319 @@
+
+\usecase{BPS}{Interact with the Data Backbone}
+The Operator wants to have an ability to use the inputs for the payloads which
+originate from the Data Backbone and store the results there.
+
+\usecase{BPS}{Dispatch and manage pipelines at scale}
+The Operator wants to dispatch and manage pipelines at the scale needed to meet
+LSST objectives within the objective’s time constraints.
+
+\usecase{BPS}{Spawn a familiar}
+The Operator wants a pet to be present during a campaign execution for their
+emotional support.
+
+\section{Monitoring}
+
+\usecase{BPS}{Monitor available resources}
+The Operator wants to know what is the state of available computational
+resources (e.g. are there any unscheduled outages).
+
+\usecase{BPS}{Monitor campaign/pipeline/job/SuperTask execution}
+The Operator or the Campaign Manager tracks campaign/pipeline/job execution,
+i.e., monitors in real time metrics such as:
+\begin{itemize}
+  \item
+    number of pending, running, finished, and failed jobs/SuperTasks;
+  \item
+    amount of computer resources (e.g. CPUs, memory, disk space) in use vs
+    idle;
+  \item
+    job runtime information (e.g. hostname, memory, wall/CPU time, data input
+    and output volume).
+  \item
+    what SuperTask or framework step is currently running (if possible seeing
+    stdout/stderr from the step) for when pipelines seem to be taking too long.
+\end{itemize}
+
+\usecase{BPS}{Verify results integrity}
+The Operator wants the ability to configure the system to verify that all the
+required outputs were generated and staged out correctly for each successful
+pipeline according to the pipeline configuration.
+
+Note: Validation that the pipeline configuration and execution produced the correct results is outside of the scope of this use case.
+
+\usecase{BPS}{Notify about events}
+The Operator may want to be notified about pipeline/campaign level events (e.g.
+its failure, completion, jobs taking longer than some operator-specified
+threshold, etc.)
+
+\section{History}
+
+\usecase{BPS}{View runtime metrics and provenance}
+For each step (including SuperTasks, scheduling, data transfer, database
+access, pre-flight, etc.), the Operator wants to view/query information such
+as:
+\begin{itemize}
+  \item
+    when it was running and time it took for the process to complete,
+  \item
+    amount of memory it used during its execution,
+  \item
+    machine where it was executed,
+  \item
+    version of software used,
+  \item
+    what input datasets were used (where applicable),
+  \item
+    what output datasets were produced (where applicable).
+\end{itemize}
+
+\usecase{BPS}{View data quality metrics}
+For each campaign/pipeline, the Operator wants to view any metrics measuring
+data quality output by the pipeline.
+
+\usecase{BPS}{Browse stdout/stderr/logs}
+The Operator wants to have easy access to stderr/stdout/logs finished jobs,
+i.e., being able to quickly find the log file of a specific job, search log
```

Comment:
Would it be easy to find the log of a specific PipelineTask too?

Reply:
Yes, the framework will allow to find log of a specific PipelineTask. I'll change wording to make it clear.